llama_index/retrievers/pathway_retriever.py

```diff
@@ -0,0 +1,168 @@
+
+        self.vector_store_server = vector_store.VectorStoreServer(
+            *docs,
+            embedder=embedding_callable,
+            parser=parser,
+            splitter=generic_transformer,
+            **kwargs,
+        )
+
+    def run_server(
+        self,
+        host: str,
+        port: str,
+        threaded: bool = False,
+        with_cache: bool = True,
+        cache_backend: Any = None,
+    ) -> Any:
+        """
+        Run the server and start answering queries.
+
+        Args:
+            host (str): host to bind the HTTP listener
+            port (str | int): port to bind the HTTP listener
+            threaded (bool): if True, run in a thread. Else block computation
+            with_cache (bool): if True, embedding requests for the same contents are cached
+            cache_backend: the backend to use for caching if it is enabled. The
+              default is the disk cache, hosted locally in the folder ``./Cache``. You
+              can use ``Backend`` class of the [`persistence API`]
+              (/developers/api-docs/persistence-api/#pathway.persistence.Backend)
+              to override it.
+
+        Returns:
+            If threaded, return the Thread object. Else, does not return.
+        """
+        try:
+            import pathway as pw
+        except ImportError:
+            raise ImportError(
+                "Could not import pathway python package. "
+                "Please install it with `pip install pathway`."
+            )
+        if with_cache and cache_backend is None:
+            cache_backend = pw.persistence.Backend.filesystem("./Cache")
+        return self.vector_store_server.run_server(
+            host,
+            port,
+            threaded=threaded,
+            with_cache=with_cache,
+            cache_backend=cache_backend,
+        )
+
+
+class PathwayRetriever(BaseRetriever):
+    """Pathway retriever.
+    Pathway is an open data processing framework.
+    It allows you to easily develop data transformation pipelines
+    that work with live data sources and changing data.
+
+    This is the client that implements Retriever API for PathwayVectorServer.
+    """
+
+    def __init__(
+        self,
+        host: str,
+        port: Union[str, int],
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        """Initializing the Pathway retriever client."""
+        import_err_msg = "`pathway` package not found, please run `pip install pathway`"
+        try:
+            from pathway.xpacks.llm.vector_store import VectorStoreClient
+        except ImportError:
+            raise ImportError(import_err_msg)
+        self.client = VectorStoreClient(host, port)
+        super().__init__(callback_manager)
+
+    def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
+        """Retrieve."""
+        rets = self.client(query=query_bundle.query_str, k=3)
```

Comment:
probably, the k should be customizable, using a familiar kwarg in the constructor like `similarity_top_k`

Reply:
Good catch, fixed.